### PR TITLE
Enabled inference engine intercompatibility

### PR DESCRIPTION
**Pull request by blindcrone**
If inference engines can run the same model, nodes on those inference engines can now interoperate

### CODE REVIEW

File: exo/download/shard_download.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ def on_progress(self) -> AsyncCallbackSystem[str, Tuple[Shard, RepoProgressEvent
 
 
 class NoopShardDownloader(ShardDownloader):
-  async def ensure_shard(self, shard: Shard) -> Path:
+  async def ensure_shard(self, shard: Shard, inference_engine_name: str) -> Path:
```
Comment (dtnewman) [inline exo/download/shard_download.py:30]: what's the intention of adding `inference_engine_name` here?

Comment (blindcrone) [reply exo/download/shard_download.py:30]: Necessary because of the change to the abstractmethod signature

Comment (dtnewman) [reply exo/download/shard_download.py:30]: Im assuming this is for some future where different inference engines save the model in different ways. Whatever the reason, maybe add it to the docstring for the abstract method `ensure_shard` above?

Comment (dtnewman) [reply exo/download/shard_download.py:31]: OK... sorry, I see what it's doing now. On first pass I didn't notice that the class in hf_shard_download.py inherits from the parent in this file :)

**Comment by AlexCheema** [comment]
This is fire

**Pull request merged**